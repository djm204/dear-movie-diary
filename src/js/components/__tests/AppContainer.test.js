import React from 'react'
import App from '../appContainer'
import 'whatwg-fetch'

import { screen, render, waitForDomChange } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

test('search for citizen kane, assert results', async () => {
  render(<App />)
  const submit = screen.getByText(/submit/i)
  const results = screen.getByText(/results/i)
  const input = screen.getByLabelText(/search for a movie../i)

  //initial
  expect(input.value).toBe('')
  expect(results.textContent).toBe('Results: {}')

  //search for citizen kane
  userEvent.type(input, 'citizen kane')
  userEvent.click(submit)
  await waitForDomChange({ container: results })

  //assert results
  expect(results.textContent).toBe('Results: ' + JSON.stringify(expectedCitizenKaneResponse))
})

const expectedCitizenKaneResponse = {
  Search: [
    {
      Title: 'Citizen Kane',
      Year: '1941',
      imdbID: 'tt0033467',
      Type: 'movie',
      Poster:
        'https://m.media-amazon.com/images/M/MV5BYjBiOTYxZWItMzdiZi00NjlkLWIzZTYtYmFhZjhiMTljOTdkXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg',
    },
    {
      Title: 'Beyond Citizen Kane',
      Year: '1993',
      imdbID: 'tt1356393',
      Type: 'movie',
      Poster:
        'https://m.media-amazon.com/images/M/MV5BMDEwMjAwZGEtZmMwNi00NmJhLWJlZjktNzg4MjdhNmVjY2U3XkEyXkFqcGdeQXVyMTkzODUwNzk@._V1_SX300.jpg',
    },
    {
      Title: "Reflections on 'Citizen Kane'",
      Year: '1991',
      imdbID: 'tt0212466',
      Type: 'movie',
      Poster: 'N/A',
    },
    {
      Title: 'The Complete Citizen Kane',
      Year: '1991',
      imdbID: 'tt2912252',
      Type: 'movie',
      Poster: 'N/A',
    },
    {
      Title: 'Citizen versus Kane',
      Year: '2009',
      imdbID: 'tt1387142',
      Type: 'movie',
      Poster:
        'https://m.media-amazon.com/images/M/MV5BMjg2ODM3MjI3NF5BMl5BanBnXkFtZTcwNDc0ODk1Mg@@._V1_SX300.jpg',
    },
    {
      Title: 'Citizen Kane and Zombies',
      Year: '2011',
      imdbID: 'tt2626774',
      Type: 'movie',
      Poster: 'N/A',
    },
    {
      Title: 'Restoring Citizen Kane',
      Year: '2012',
      imdbID: 'tt5733042',
      Type: 'movie',
      Poster: 'N/A',
    },
  ],
  totalResults: '7',
  Response: 'True',
}
