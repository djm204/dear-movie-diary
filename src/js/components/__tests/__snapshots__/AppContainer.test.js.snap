// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search for citizen kane, assert results 1`] = `
<div
  class="SearchResults"
  data-testid="search-results"
>
  <div
    class="SearchResults_ConfirmFavorites"
  >
    <form>
      <a
        href="/confirmation"
      >
        <button
          class="Button disabled"
          disabled=""
          type="submit"
        >
          Confirm Favorites
        </button>
      </a>
    </form>
  </div>
  <div
    class="SearchResults_Data"
  >
    <div
      class="MovieThumbnail"
      data-imdbid="tt0033467"
      data-testid="Citizen Kane1941"
    >
      <img
        alt="Citizen Kane"
        class="MovieThumbnail_Image"
        src="https://m.media-amazon.com/images/M/MV5BYjBiOTYxZWItMzdiZi00NjlkLWIzZTYtYmFhZjhiMTljOTdkXkEyXkFqcGdeQXVyNzkwMjQ5NzM@._V1_SX300.jpg"
      />
      <div
        class="MovieThumbnail_AddToFavorites"
      >
        <div
          class="MovieThumbnail_AddToFavorites_Message"
        >
          Add to Favorites
        </div>
      </div>
    </div>
    <div
      class="MovieThumbnail"
      data-imdbid="tt1356393"
      data-testid="Beyond Citizen Kane1993"
    >
      <img
        alt="Beyond Citizen Kane"
        class="MovieThumbnail_Image"
        src="https://m.media-amazon.com/images/M/MV5BMDEwMjAwZGEtZmMwNi00NmJhLWJlZjktNzg4MjdhNmVjY2U3XkEyXkFqcGdeQXVyMTkzODUwNzk@._V1_SX300.jpg"
      />
      <div
        class="MovieThumbnail_AddToFavorites"
      >
        <div
          class="MovieThumbnail_AddToFavorites_Message"
        >
          Add to Favorites
        </div>
      </div>
    </div>
    <div
      class="MovieThumbnail"
      data-imdbid="tt0212466"
      data-testid="Reflections on 'Citizen Kane'1991"
    >
      <div
        class="MovieThumbnail_Image --no-image"
      >
        Reflections on 'Citizen Kane'
         - 
        1991
      </div>
      <div
        class="MovieThumbnail_AddToFavorites"
      >
        <div
          class="MovieThumbnail_AddToFavorites_Message"
        >
          Add to Favorites
        </div>
      </div>
    </div>
    <div
      class="MovieThumbnail"
      data-imdbid="tt2912252"
      data-testid="The Complete Citizen Kane1991"
    >
      <div
        class="MovieThumbnail_Image --no-image"
      >
        The Complete Citizen Kane
         - 
        1991
      </div>
      <div
        class="MovieThumbnail_AddToFavorites"
      >
        <div
          class="MovieThumbnail_AddToFavorites_Message"
        >
          Add to Favorites
        </div>
      </div>
    </div>
    <div
      class="MovieThumbnail"
      data-imdbid="tt1387142"
      data-testid="Citizen versus Kane2009"
    >
      <img
        alt="Citizen versus Kane"
        class="MovieThumbnail_Image"
        src="https://m.media-amazon.com/images/M/MV5BMjg2ODM3MjI3NF5BMl5BanBnXkFtZTcwNDc0ODk1Mg@@._V1_SX300.jpg"
      />
      <div
        class="MovieThumbnail_AddToFavorites"
      >
        <div
          class="MovieThumbnail_AddToFavorites_Message"
        >
          Add to Favorites
        </div>
      </div>
    </div>
    <div
      class="MovieThumbnail"
      data-imdbid="tt2626774"
      data-testid="Citizen Kane and Zombies2011"
    >
      <div
        class="MovieThumbnail_Image --no-image"
      >
        Citizen Kane and Zombies
         - 
        2011
      </div>
      <div
        class="MovieThumbnail_AddToFavorites"
      >
        <div
          class="MovieThumbnail_AddToFavorites_Message"
        >
          Add to Favorites
        </div>
      </div>
    </div>
    <div
      class="MovieThumbnail"
      data-imdbid="tt5733042"
      data-testid="Restoring Citizen Kane2012"
    >
      <div
        class="MovieThumbnail_Image --no-image"
      >
        Restoring Citizen Kane
         - 
        2012
      </div>
      <div
        class="MovieThumbnail_AddToFavorites"
      >
        <div
          class="MovieThumbnail_AddToFavorites_Message"
        >
          Add to Favorites
        </div>
      </div>
    </div>
  </div>
</div>
`;
